src/main/java/org/breedinginsight/brapi/v2/dao/ProgramCache.java

```diff
@@ -31,19 +31,19 @@
 import java.util.stream.Collectors;
 
 @Slf4j
-public class ProgramCache<R> {
+public class ProgramCache<K, R> {
 
-    private final FetchFunction<UUID, List<R>> fetchMethod;
+    private final FetchFunction<UUID, Map<K, R>> fetchMethod;
```

Comment:
Not clearly explained that K should be the object id, it took me a bit to figure out what K was supposed to be. Maybe a class header comment for explanation?